index_configuration_tool/tests/test_logstash_parser.py

```diff
@@ -0,0 +1,76 @@
+import pickle
+import unittest
+from os.path import dirname
+
+import lark.exceptions
+from jsondiff import diff
+
+from logstash_conf_parser import logstash_parser, parse
+
+# Constants
+LOGSTASH_TEST_INPUT_FILE = dirname(__file__) + "/logstash_test_input.conf"
+LOGSTASH_EXPECTED_OUTPUT_FILE = dirname(__file__) + "/expected_parse_output.pickle"
+
+
+class TestLogstashParser(unittest.TestCase):
+    # Run before each test
+    def setUp(self) -> None:
+        with open(LOGSTASH_EXPECTED_OUTPUT_FILE, "rb") as f:
+            # The root DS is a dict, with input type as key.
+            # The value of each key is an array of inputs.
+            # Each input is a tuple of plugin name and data,
+            # where the data is a dict of key-value pairs.
+            self.test_data = pickle.load(f)
+
+    # Test input json should match loaded pickle data
+    def test_parser_happy_case(self):
+        actual = parse(LOGSTASH_TEST_INPUT_FILE)
+        test_diff = diff(self.test_data, actual)
+        # Validate that diff is empty
+        self.assertFalse(test_diff)
+
+    def test_bad_configs(self):
+        with self.assertRaises(lark.exceptions.UnexpectedToken):
+            # Empty config is an error
+            logstash_parser.parse("")
+            # Sections should begin with type name
+            logstash_parser.parse("{}")
+            # Invalid type
+            logstash_parser.parse("bad {}")
+            # Valid type but without params
+            logstash_parser.parse("input")
+
+    def test_empty_config(self):
+        logstash_parser.parse("input {}")
```

Comment:
To be clear, these are valid and should pass? Maybe worth a comment or naming the function as `test_empty_config_is_valid` or something -- they _look_ pretty invalid. 

Actually, these succeed in parsing, but would fail the `validate_logstash_config` call, right? So maybe `test_empty_config_can_be_parsed` or along those lines?

Reply:
Changed to `test_empty_config_can_be_parsed` and added a comment